packages/edit-site/src/components/page-content-lock/use-disable-non-content-blocks.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * WordPress dependencies
+ */
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { addFilter, removeFilter } from '@wordpress/hooks';
+import {
+	store as blockEditorStore,
+	privateApis as blockEditorPrivateApis,
+} from '@wordpress/block-editor';
+import { useSelect } from '@wordpress/data';
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../../private-apis';
+
+const { useBlockEditingMode } = unlock( blockEditorPrivateApis );
+
+/**
+ * Disables non-content blocks using the `useBlockEditingMode` hook.
+ */
+export function useDisableNonContentBlocks() {
```

Comment:
Might the block-editor package be a better home for this? It would be super useful to have this template/content distinction for post editor template editing too (are there plans to do that at all?)

Reply:
Right now we only surface pages in the site editor via the W menu. I could see us eventually surfacing posts as well, even if it's only via navigating your site in browse mode on the right hand side. Even more long term I could see the site editor replacing the post editor.

This would all still be in the `edit-site` package, though, so having `useDisableNonContentBlocks` in `block-editor` wouldn't help in any way.

Reply:
From my perspective even today this api could be used to load a preview of the template in the post editor like described in #27847 

So even before an eventual unification I think there is great value in this starting now for a very large user base of every WordPress user that isn't an administrator. Because currently the site editor is limited to administrators.

Reply:
I feel that the `edit-site` editor is a much better starting place for building a unified editor (one that can switch between posts and templates simultaneously) than the `edit-post` editor. The `edit-site` editor is already able to handle multiple entity types and the `edit-post` editor contains a lot of legacy stuff e.g. custom fields. In other words I think a better alternative to https://github.com/WordPress/gutenberg/issues/27847 is to expand the functionality we're adding in this PR to the point where we can remove the post editor entirely in lieu of the site editor for block themes. As part of that we would have to make `site-editor.php` more intelligent about what capabilities it requires so that non-admins can use it.

But getting ahead of myself. This is all very long term and to relevant to this fairly limited feature we're starting with.

It's far easier to move something from `edit-site` to `block-editor` in the future than the other way around due to backwards compatibility concerns, so I'm comfortable with `useDisableNonContentBlocks` starting out here.